app/controllers/components/course/materials_component.rb

```diff
@@ -0,0 +1,15 @@
+class Course::MaterialsComponent < SimpleDelegator
+  include Course::ControllerComponentHost::Component
+
+  def sidebar_items
+    [
+      {
+        key: :materials,
+        title: I18n.t('course.material.sidebar_title'),
+        weight: 4,
+        path: course_material_folder_path(current_course, current_course.root_folder),
```

Comment:
To think further, if changes are complicated, we probably will use Models, otherwise I don't think we can make it.

So I would still prefer using a rake task, but not commit to git, They are mostly temporary tasks. new deployment won't need to run such tasks.